rails/es6.md

```diff
@@ -0,0 +1,105 @@
+## [Home](../README.md)
+
+## Summary
+
+Including ES6 in Rails applications can be very useful. Using JavaScript
+packages that are only available/maintained in npm version has traditionally
+been a problem with Rails application. This documents an approach chosen among a
+couple of different options.
+
+We settled upon sprockets-commoner over webpack and browserify because it
+integrates with the Rails asset pipeline and least modifies the natural
+development of JavaScript components.
+
+### Steps to upfit your Rails application
+  * add `'expose window.[component_name]'` to the top of ES6 class files
+  * add `gem 'sprockets-commoner'` to Gemfile
+  * add `.babelrc` to project root
+    * add other presets as necessary (eg. react)
+
+  ```json
+    {
+      presets: [
+        "es2015"
+      ]
+    }
+  ```
+  * add NPM dependencies to `package.json` (including react-dom, react-chartjs, etc.)
+
+  ```json
+    "dependencies": {
+      "autoprefixer": "^6.3.5",
+      "axios": "^0.9.1",
```

Comment:
That's probably true. We added in some other ones that were needed by our example application. I pulled a couple of them out from this list but did not go through the whole list. We opted to leave additional ones in rather than incorrectly remove a necessary one. Someone who knows these dependencies better might know better which should remain.